app/assets/javascripts/update.js

```diff
@@ -0,0 +1,34 @@
+function editIdeaListener(){
```

Comment:
So the problem here, I believe, is that you don't call the editIdeaListener() function anywhere in the code - so this event listener never gets set in the code!